### PR TITLE
Update documentation

### LINKED ISSUE #3661: docs: Issue in page Examples
I would like to report an issue in page https://manager.docs.scylladb.com/branch-3.2/restore/examples

### Problem
For a restore to a different cluster, what cluster name should be used?  The name of the source cluster that produced the backup, or the name of the target cluster?
> where:
> -c - the [name](https://manager.docs.scylladb.com/stable/sctool/cluster.html#cluster-add) you used when you created the cluster

###  Suggest a fix

Be more specific about which cluster name is needed for different restore scenarios.

### LINKED ISSUE #3663: docs: Issue in page Restore tables
I would like to report an issue in page https://manager.docs.scylladb.com/branch-3.2/restore/restore-tables

### Problem
When I read the "process" section of this doc, my first impression is - the docs want me to do those things.
I realize now - the restore process is going to do these things.  But I went through and did the schema alters because it felt like it wanted me to do that...  (I also realize that since I altered them, I will have to alter them back after the restore is complete)

###  Suggest a fix

Revise the section to be more clear that "Scylla Manager Restore will perform the following actions"

### LINKED ISSUE #3681: Add a note to follow Scylla Operator docs for relevant procedures
Manager docs should add a note to each procedure to follow a dedicated docs for the operator instead, for a supported procedure and specific instructions.

At this point I can think of:
- backup
- restore
- configuring repairs

But it applies in general.

### LINKED ISSUE #3684: Restore examples docs are outdated
As per https://github.com/scylladb/scylla-operator/pull/1678#discussion_r1445874550, the [restore examples doc](https://manager.docs.scylladb.com/stable/restore/examples.html) is outdated and needs to be aligned with the up-to-date procedure.

cc @Michal-Leszczynski

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
This PR fixes a bunch of documentation related issues.

Fixes #3684
Fixes #3681
Fixes #3661
Fixes #3663

### CODE REVIEW

File: docs/source/backup/index.rst [hunk 1]
```diff
@@ -19,6 +19,9 @@ Backup
 
 Using :ref:`sctool backup <sctool-backup>` command, you can schedule a backup of a managed cluster.
 Backups and repairs are scheduled in the same manner, you can start, stop, resume, and track task progress on demand.
+
+.. note:: If you are using ScyllaDB Manager deployed by ScyllaDB Operator, see dedicated `ScyllaDB Operator documentation <https://operator.docs.scylladb.com/stable/manager.html>`_.
```
Comment (tzach) [inline docs/source/backup/index.rst:23]: FYI, you can use [RST. Include](https://docutils.sourceforge.io/docs/ref/rst/directives.html#including-an-external-document-fragment) or repeat sections. It's not required here.

**Review by tzach — APPROVED** [review]
LGTM

**Review by karol-kokoszka — APPROVED** [review]

**Comment by Michal-Leszczynski** [comment]
@annastuchlik could you take a look at this PR?

**Comment by Michal-Leszczynski** [comment]
The problem with using include for linking to operator docs is that there are separate pages for repair/backup and for restore. So for now I will keep the links directly in the code, so that they don't get confused, but it can be changed later on.

**Pull request merged**